config.go

```diff
@@ -37,7 +37,8 @@ type Config struct {
 	// Shell configures the devbox shell environment.
 	Shell struct {
 		// InitHook contains commands that will run at shell startup.
-		InitHook ConfigShellCmds `json:"init_hook,omitempty"`
+		InitHook ConfigShellCmds             `json:"init_hook,omitempty"`
+		Scripts  map[string]*ConfigShellCmds `json:"scripts,omitempty"`
```

Comment:
in `launchpad.yaml`, we had a similar undeterministic-ordering issue with the Services. We handled it by:
1. defining it as `Scripts []ConfigShellCmds` and adding the "name" as a field in `ConfigShellCmds`
2. adding custom marshal and unmarshal functions to have it print as a map with the key

Would you wanna try that? Could do as a followup. I feel it would be really nice to ensure determinism. With determinism, we won't get random diffs of devbox.json when we write to it for (other unrelated) reasons.

Reply:
oh interesting, i'll see about it!

Reply:
I'll do as a follow-up